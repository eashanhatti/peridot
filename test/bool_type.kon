let {
	datatype Bool : OType {
		true  : Bool;
		false : Bool;
	};
} in {
	(pure unit)
}
