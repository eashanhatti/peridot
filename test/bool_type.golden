((fromList [(Global {unGlobal = 0}, Just UnitType)],
  fromList [],
  fromList [],
  fromList []),
 Let
   (fromList
      [ObjectConstant (Id {unId = 0}) (TypeType Object),
       ObjectConstant
         (Id {unId = 1}) (GlobalVar (Id {unId = 0})),
       ObjectConstant
         (Id {unId = 2}) (GlobalVar (Id {unId = 0}))])
   (IOIntroPure UnitIntro))
